docs/use_cases/retrieval_augmented_generation_eval_qdrant_ragas.md

```diff
@@ -0,0 +1,643 @@
+   Your goal is to provide a clear and concise response that addresses their query while referencing relevant information 
+   from the documentation.
+   Remember to: 
+   Understand the user's question thoroughly.
+   If the user's query is general (e.g., "hi," "good morning"), 
+   greet them normally and avoid using the context from the documentation.
+   If the user's query is specific and related to the documentation, locate and extract the pertinent information.
+   Craft a response that directly addresses the user's query and provides accurate information
+   referring the relevant source and page from the 'source' field of fetched context from the documentation to support your answer.
+   Use a friendly and professional tone in your response.
+   If you cannot find the answer in the provided context, do not pretend to know it.
+   Instead, respond with "I don't know".
+   
+   Context:\n"""
+)
+
+prompt_end = (
+   f"\n\nQuestion: {query}\nAnswer:"
+)
+
+prompt = (
+        prompt_start + "\n\n---\n\n".join(contexts) +
+        prompt_end
+)
+
+res = openai_client.completions.create(
+   model="gpt-3.5-turbo-instruct",
+   prompt=prompt,
+   temperature=0,
+   max_tokens=636,
+   top_p=1,
+   frequency_penalty=0,
+   presence_penalty=0,
+   stop=None
+)
+
+return (contexts , res.choices[0].text)
+```
+
+Try out running your first query through your RAG pipeline as :
+
+```python
+question1 = "what is quantization?"
+RETRIEVAL_WINDOW_SIZE_5 = 5
+context1 , rag_response1 = query_with_context(question1,RETRIEVAL_WINDOW_SIZE_5)
+#print(context)
+print(rag_response1)
+
+##Outputs 
+#Quantization is an optional feature in Qdrant that enables efficient storage and search of high-dimensional vectors. 
+# It compresses data while preserving relative distances between vectors, making vector search algorithms more efficient. 
+# This is achieved by converting traditional float32 vectors into qbit vectors and creating quantum entanglement between them. 
+# This unique phenomenon in quantum systems allows for highly efficient vector search algorithms. 
+# For more information on quantization methods and their mechanics, please refer to the documentation on quantization at https://qdrant.tech/documentation/guides/quantization.html.
+```
+
+One key thing to understand is that RAGAS work inherently with datasets for evaluations for it is a good idea to create a set of questions and ground truth to help create an evaluation dataset to assess our RAG system. We are leveraging the dataset [https://huggingface.co/datasets/atitaarora/qdrant_doc_qna](https://huggingface.co/datasets/atitaarora/qdrant_doc_qna) to create our evaluation set from our RAG pipeline and execute it as shown [before](https://www.notion.so/Evaluating-RAG-with-RAGAS-ef0de436a00748c7bba9fd7161cd76cc?pvs=21).
+
+**Lets evaluate , how good is our RAG !!**
+
+Its as simple as importing the desired metrics:
+
+```python
+from ragas.metrics import (
```

Comment:
@atarora 
here you have 5 metrics, whereas ragas-score was supposed to be the mean of the 4 main metrics

Reply:
true , I do not have `Context entity recall` , `Context relevancy` and `Answer semantic similarity` here , I can add them and regenerate the metrics , update code and comparison screenshot.

Please remember - 4 aspects and 8 metrics :)

Reply:
4 aspects, 8 metrics
what are the four aspects? And which metrics correspond to which aspects?
As far as I can tell, the 4 aspects = the first 4 metrics (from line 278). Then there are 4 more. (edited)

Reply:
plus, ragas-score is the mean of 4 metrics, not aspects. The 4 aspects of RAGAS are 1 data input, 2 metric selection, 3 evaluation process, and 4 result interpretation. I don't see any discussiong of these. My question about is about ragas-score (ie metrics).

Reply:
I can expand on the aspects before I cover metrics and their details and I think you may have a point that the picture - https://docs.ragas.io/en/latest/concepts/metrics/index.html (sadly the one used everywhere) isn't quite a representation of the aspects - retrieval , generation , end to end evaluation and critique (infact that is the newest introduced in the version released after this article was written). My code notebook reflects these but may be this article may need to be expanded to add it. 
Do you suggest I talk about `aspects` before describing metrics ?